nixos/modules/services/web-apps/calibre-web.nix

```diff
@@ -0,0 +1,162 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.calibre-web;
+
+  inherit (lib) concatStringsSep mkEnableOption mkIf mkOption optional types;
+in
+{
+  options = {
+    services.calibre-web = {
+      enable = mkEnableOption "Calibre-Web";
+
+      listen = {
+        ip = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = ''
+            IP address that Calibre-Web should listen on.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 8083;
+          description = ''
+            Listen port for Calibre-Web.
+          '';
+        };
+      };
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "calibre-web";
+        description = ''
+          The directory below <filename>/var/lib</filename> where Calibre-Web stores its data.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "calibre-web";
+        description = "User account under which Calibre-Web runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "calibre-web";
+        description = "Group account under which Calibre-Web runs.";
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Open ports in the firewall for the server.
+        '';
+      };
+
+      settings = {
```

Comment:
if the option is called settings, it should respect RFC42 https://github.com/NixOS/rfcs/blob/45b76f20add8d7e0e1d0bc0eed9357b067c899a6/rfcs/0042-config-option.md

It looks like the format for the configuration file is INI, so it should be easy.

Reply:
I'll have a look at the RFC. But just FYI calibre-web doesn't have a configuration file. The settings listed in the module are written into sqlite DB by systemd pre-exec script (it's the only way to configure calibre-web).

Reply:
Oh indeed I had not seen the string constructed line 111 was fed to sqlite3. Well in this case I'd suggest renaming the option to something else than settings, just to avoid suggesting this is a RFC42-style settings option.